### PR TITLE
fix: fix downloaded CSV votes file not containing all votes

### LINKED ISSUE #1626: Bring back the feature to download all votes as an CSV file

### PR DESCRIPTION
**Pull request by wa0x6e**
### Issues
When downloading the CSV files for a proposal votes, the file is capped at 6000 rows due to implementation of the new limit on the API side.

Fixes #1626

### Changes 
Implement the changes, as described in https://discord.com/channels/707079246388133940/1061540232769577061/1061540232769577061

1. the proposal query is now using the created_gte condition to get over the 6000 rows limit
2. the data in the CSV file is now ordered by created, instead of voting power
3. A progress icon is shown while the CSV file is being built

### How to test

1. Go to a proposal with more than 6k votes (such as https://snapshot-git-fix-1626-snapshot.vercel.app/#/aave.eth/proposal/0xf6da417983653f7e7f5560a8e16ad77d5e25533bba9157abfd222347b40d09d0?app=senate)
4. Download all the votes as CSV file
5. A progress icon should be shown in lieu of the button, to track the progress
6. The file should contains as many rows as the votes count
7. The columns for the choices are adjacent, and order asc

### To-Do

- [x] Handle duplicate votes
- [x] missing comma before address, and missing line break
- [x] vote choice column is not ordered

### Self-review checklist
- [x] I have performed a full self-review of my changes
- [x] I have tested my changes on a preview deployment
- [ ] I have tested my changes on a custom domain
- [ ] I have run end-to-end tests `yarn cypress:test:e2e`, and they have passed

### Additional notes or considerations

Data are now sorted by created. We could also revert back to `voting power` if desired, but it may be computation heavy, as the sorting is done on the browser side.

### CODE REVIEW

**Comment by samuveth** [comment]
@wa0x6e could you also look into 2. And 3. from the screenshot here: https://discord.com/channels/955773041898573854/1030772384308932608/1095560019489063013

File: src/composables/useReportDownload.ts [hunk 1]
```diff
@@ -1,42 +1,74 @@
...
     let votes: Vote[] = [];
     let page = 0;
+    let createdPivot = 0;
     const pageSize = 1000;
-    while (votes.length === pageSize * page) {
-      const newVotes = await getProposalVotes(proposalId, {
+    let resultsSize = 0;
+    const maxPage = 5;
+    do {
+      let newVotes = await getProposalVotes(proposalId, {
```
Comment (samuveth) [inline src/composables/useReportDownload.ts:35]: Should we use a try catch here?

File: src/locales/default.json [hunk 2]
```diff
@@ -250,7 +250,9 @@
       "tips": {
         "1": "Votes can be changed while the proposal is active"
       }
-    }
+    },
+    "downloadCsvVotes": "Download as CSV",
+    "preparingCsvVotes": "Preparing the CSV file"
```
Comment (samuveth) [inline src/locales/default.json:255]:
```suggestion
    "preparingCsvVotes": "Preparing file"
```

Comment (samuveth) [reply src/locales/default.json:255]: It's a bit long

File: src/composables/useReportDownload.ts (hunk 1 above)
Comment (wa0x6e) [reply src/composables/useReportDownload.ts:35]: There's already a try/catch on the `getProposalVotes` side.

We could remove it, and move it to the UI side, to handle errors

Comment (wa0x6e) [reply src/composables/useReportDownload.ts:35]: That will make the error bubble 2 layers upward, snapshot.ts > userReportDownload.ts > SpaceProposalVotesList.vue, and also require all components using `getProposalVotes` to implement their own try/catch

Comment (samuveth) [reply src/composables/useReportDownload.ts:35]: Yes right, let's leave it as is I'd say

File: src/components/SpaceProposalVotesList.vue [hunk 3]
```diff
@@ -44,14 +44,15 @@ onMounted(async () => {
       >
         <i-ho-download />
       </BaseButtonIcon>
-      <BaseButtonIcon
+      <div
```
Comment (samuveth) [inline src/components/SpaceProposalVotesList.vue:47]: This you forgot to remove the props and a a loader?

**Comment by samuveth** [comment]
Thanks @wa0x6e

**Pull request merged**